### PR TITLE
DAOS-9987 build: Do not skip centos 7 for quick-builds.

### PR DESCRIPTION
**Pull request by ashleypittman**
Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: vars/skipStage.groovy [hunk 1]
```diff
@@ -213,8 +213,7 @@ boolean call(Map config = [:]) {
             return paramsValue('CI_BUILD_PACKAGES_ONLY', false) ||
                    skip_stage_pragma('build-centos7-gcc', 'false') ||
                    (docOnlyChange(target_branch) &&
-                    prRepos('centos7') == '') ||
-                   quickBuild()
+                    prRepos('centos7') == '')
```
Comment (brianjmurrell) [inline vars/skipStage.groovy:216]:
```suggestion
                    prRepos('centos7') == '') ||
                    quickFunctional()
```

Is likely what was originally intended.  QB runs UT but QF skips UT and therefore non-RPM builds are not necessary.

And yes, this entire stage-skipping code needs a redesign.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: vars/skipStage.groovy (hunk 1 above)
Comment (ashleypittman) [reply vars/skipStage.groovy:216]: I don't think I put that much effort into what was intended.  Whilst we're here what would be required to add checkpatch (and presumably pylint) back into quickFunctional()

Comment (brianjmurrell) [reply vars/skipStage.groovy]: We've already had this discussion.  `Quick-functional`'s entire point of existence is to get to functional testing as quickly as possible.  Do not pass go, do not collect $100.  If you want functional only testing but with linting and checkpatch and whatnot (i.e. a functional-only testing path to landing), that is a new pragma.  Quick-functional's purpose is to iterate on functional testing as quickly as possible.  It's mean for development only.

**Review by brianjmurrell — APPROVED** [review]

File: vars/skipStage.groovy (hunk 1 above)
Comment (ashleypittman) [reply vars/skipStage.groovy]: Yeah, and I got caught by it again last week where I waited 48 hours for a test-run only to find out I'd got the syntax wrong.

**Comment by jolivier23** [comment]
what's up with all of the failures?

**Comment by brianjmurrell** [comment]
> > what's up with all of the failures?
> 
> No idea. Build 3 was fine but a subsequent rebase failed. I filed a ticket about PR builds failing for pipeline-lib recently at https://daosio.atlassian.net/browse/CORCI-1154

They look to be related to the new go packages that we are building ourselves:

```
[2022-03-02T16:48:47.565Z] Error: Package: golang-bin-1.16.13-2.el7.x86_64 (repo.dc.hpdd.intel.com_repository_epel-el-7-x86_64-proxy-fixed_)
[2022-03-02T16:48:47.565Z]            Requires: go = 1.16.13-2.el7
[2022-03-02T16:48:47.565Z]            Removing: golang-1.15.14-1.el7.x86_64 (@repo.dc.hpdd.intel.com_repository_epel-el-7-x86_64-proxy-fixed_)
[2022-03-02T16:48:47.565Z]                go = 1.15.14-1.el7
[2022-03-02T16:48:47.565Z]            Updated By: golang-1.17.7-2.daos.el7.x86_64 (repo.dc.hpdd.intel.com_repository_daos-stack-el-7-x86_64-stable-local)
[2022-03-02T16:48:47.565Z]                go = 1.17.7-2.daos.el7
[2022-03-02T16:48:47.565Z]            Available: go-1.17.7-1.daos.el7.x86_64 (repo.dc.hpdd.intel.com_repository_daos-stack-el-7-x86_64-stable-local)
[2022-03-02T16:48:47.565Z]                go = 1.17.7-1.daos.el7
[2022-03-02T16:48:47.565Z]            Available: golang-1.16.13-2.el7.x86_64 (repo.dc.hpdd.intel.com_repository_epel-el-7-x86_64-proxy-fixed_)
[2022-03-02T16:48:47.565Z]                go = 1.16.13-2.el7
```

**Comment by brianjmurrell** [comment]
> Test stage provisionNodes with slurm Leap15 completed with status FAILURE. https://build.hpdd.intel.com/job/daos-stack/job/pipeline-lib/job/PR-288/12/display/redirect

Looks like Nexus is broken again.  Getting @JohnMalmberg to kick it.

**Pull request merged**